src/suffix.rs

```diff
@@ -209,15 +211,23 @@ impl SuffixScheme for AppendTimestamp {
     fn rotate_file(
         &mut self,
         _basepath: &Path,
+        content_limit: &ContentLimit,
         newest_suffix: Option<&TimestampSuffix>,
         suffix: &Option<TimestampSuffix>,
     ) -> io::Result<TimestampSuffix> {
         assert!(suffix.is_none());
         if suffix.is_none() {
-            let now = Local::now().format(self.format).to_string();
+            let mut now = Local::now();
+
+            match content_limit {
+                ContentLimit::Date => now = now - Duration::days(1),
```

Comment:
Why is this necessary? Just seems a bit ad hoc, I would try to think of a more general fix.